src/ryu_faucet/org/onfsdn/faucet/gauge.py

```diff
@@ -108,6 +143,17 @@ def handler_connect_or_disconnect(self, ev):
             return
 
         if ev.enter: # DP is connecting
+            # Update db with switch
+            if self.db_enabled:
+                rows = self.switch_database.get_docs(
+                    self.db_conf_data['views']['v1'],
+                    key=str(hex(ryudp.id))
+                )
+                if not rows:
+                    switch_object = {'_id': str(hex(ryudp.id)),
+                                     'data':{'flows':[]}}
+                    self.switch_database.insert_update_doc(switch_object,
+                                                           'data')
```

Comment:
this should be a watcher's job. This might take some modification to get this actually to work tidily, but I kinda suspect it should be fairly straightforward.